File: Algorithms/1630_Arithmetic_Subarrays/Solution.kt
class Solution {
    fun checkArithmeticSubarrays(nums: IntArray, l: IntArray, r: IntArray): List<Boolean> = l.zip(r)
            .map { nums.slice(it.first..it.second).sorted() }
            .map { it.zip(it.slice(1 until it.size)) }
            .map { xs -> xs.map { it.first - it.second }.distinct().count() == 1 }
}

File: Algorithms/1630_Arithmetic_Subarrays/Solution.scala
object Solution {
  def checkArithmeticSubarrays(nums: Array[Int], l: Array[Int], r: Array[Int]): List[Boolean] = l.zip(r)
    .map(xs => nums.slice(xs._1, xs._2 + 1).sorted)
    .map(xs => xs.zip(xs.slice(1, xs.length))
      .map(ys => ys._1 - ys._2)
      .distinct
      .length == 1
    ).toList
}